docs/configuration.md

```diff
@@ -41,3 +41,25 @@ A list of key-value pairs defining a custom [Pygment Style](https://pygments.org
 }
 ```
 
+### Maximum Context
+
+To save costs on larger models you can set a maximum context.
+```json
+{
+    "maximum-context": "16000"
```

Comment:
Could it make sense to default to None, or 32k? I don't want someone to change the model to gpt-4-32k but then not realize or know why it's not using the full context. If someone runs a local model or something we can't infer context length for and this isn't set we can tell them it's required

Reply:
Yeah, this is just an example for the doc. The default is `null`. There is code to set the `context_size` for gpt-3 and gpt-4 models. If the user runs a model for which we can't infer the context length and they don't set one then mentat fails with a pretty descriptive error (from openai not us) when they run out of context.

Reply:
oops haha perfect!

Reply:
> If the user runs a model for which we can't infer the context length and they don't set one then mentat fails with a pretty descriptive error (from openai not us) when they run out of context.

might be good to just require them to set it when we can't infer it, since our logic for filling up context (with code maps, embeddings, etc) will probably be based on a max. And it will save them a wasted call / get better warnings as they get close